sqlglot/parser.py

```diff
@@ -2985,8 +2999,18 @@ def _parse_column(self) -> t.Optional[exp.Expression]:
                 field = self._parse_types()
                 if not field:
                     self.raise_error("Expected type")
-            elif op:
+            elif op and self._curr:
```

Comment:
Without `self._curr` here we might raise an index error in the `self._advance` call below (happened earlier).